src/Shared/PlatformNegotiation.cs

```diff
@@ -35,12 +35,9 @@ internal static string GetNearestPlatform(string referencedProjectPlatform, stri
                 }
 
                 string buildProjectReferenceAs = string.Empty;
-
-                // If the referenced project has a defined `Platform` that's compatible, it will build that way by default.
-                // Don't set `buildProjectReferenceAs` and the `_GetProjectReferencePlatformProperties` target will handle the rest.
                 if (!string.IsNullOrEmpty(referencedProjectPlatform) && referencedProjectPlatform.Equals(currentProjectPlatform, StringComparison.OrdinalIgnoreCase))
                 {
-                    log?.LogMessageFromResources(MessageImportance.Low, "GetCompatiblePlatform.ReferencedProjectHasDefinitivePlatform", projectPath, referencedProjectPlatform);
+                    buildProjectReferenceAs = currentProjectPlatform;
```

Comment:
We should keep this part of the logic the same.

Context: Michael considered making this just set `buildProjectReferenceAs` and catching it with the "catch all" if statement below. I think it's important to preserve the logic that existed before, since there's no meaningful difference here.

Reply:
reverted